src/v/cluster/metrics_reporter.cc

```diff
@@ -101,6 +105,53 @@ address parse_url(const ss::sstring& url) {
     }
     return ret;
 }
+
+ss::sstring get_hostname() {
+    std::array<char, HOST_NAME_MAX> hostname{};
+    if (::gethostname(hostname.data(), hostname.size()) != 0) {
+        return {};
+    }
+
+    return hostname.data();
+}
+
+ss::sstring get_domainname() {
+    std::array<char, HOST_NAME_MAX> domainname{};
+    if (::getdomainname(domainname.data(), domainname.size()) != 0) {
+        return {};
+    }
+
+    return domainname.data();
+}
+
+std::vector<ss::sstring> get_fqdn(std::string_view hostname) {
+    struct addrinfo hints = {};
+    struct addrinfo* info = nullptr;
+
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_CANONNAME;
+
+    auto free_info = ss::defer([&info] {
+        if (info != nullptr) {
+            freeaddrinfo(info);
+        }
+    });
+
+    if (getaddrinfo(hostname.data(), nullptr, &hints, &info) != 0) {
```

Comment:
Any reason not to use the seastar APIs here?

Reply:
Which one?  `get_host_by_name`?

Reply:
Yeah. `getaddrinfo` is a blocking API and could cause a reactor stall.